### PR TITLE
[PR] Add support for external announcement sources

### PR DESCRIPTION
**Pull request by jeremyfelt**

### CODE REVIEW

File: includes/wsu-content-type-announcement.php [hunk 1]
```diff
@@ -125,6 +127,42 @@ public function enqueue_admin_scripts() {
...
+				$url = esc_url( 'http://' . $url_parts['host'] . $url_parts['path'] );
+			} else {
+				$url = esc_url( $url_parts['scheme'] . '://' . $url_parts['host'] . $url_parts['path'] );
+			}
+		} else {
+			$url = '';
+		}
+
+		?>
+		<p>Enter the full URL of an announcement here if the posted elsewhere.</p>
```
Comment (philcable) [inline includes/wsu-content-type-announcement.php]: This label seems to be suffering from having been edited mid-type :)

`... if it is posted elsewhere.` might convey the meaning well enough ¯\\\_(ツ)\_/¯

File: includes/wsu-content-type-announcement.php [hunk 2]
```diff
@@ -125,6 +127,42 @@ public function enqueue_admin_scripts() {
...
+		<label for="announcement-form-url">URL:</label>
+		<input type="text"
+				id="announcement-form-url"
+				name="announcement_url"
+				style="width: 575px; max-width: 100%;"
```
Comment (philcable) [inline includes/wsu-content-type-announcement.php]: Without having tried it, `width:100%;` on its own might work here just as well.

File: includes/wsu-content-type-announcement.php (hunk 1 above)
Comment (jeremyfelt) [reply includes/wsu-content-type-announcement.php]: haha, I think you're right about that. I'll update.

File: includes/wsu-content-type-announcement.php (hunk 2 above)
Comment (jeremyfelt) [reply includes/wsu-content-type-announcement.php]: The initial with of 575 makes the input appear to the right of the label until the screen is small enough, then it drops below. Not entirely necessary, but lazy formatting :)

Comment (philcable) [reply includes/wsu-content-type-announcement.php]: 👍

**Pull request merged**